FirebaseDatabase/Sources/Core/FWriteTree.m

```diff
@@ -256,14 +256,14 @@ - (NSArray *)removeAllWrites {
                 return nil;
             } else {
                 BOOL (^filter)(FWriteRecord *) = ^(FWriteRecord *record) {
-                  return (BOOL)(
-                      (record.visible || includeHiddenWrites) &&
-                      (writeIdsToExclude == nil ||
-                       ![writeIdsToExclude
-                           containsObject:
-                               [NSNumber numberWithInteger:record.writeId]]) &&
-                      ([record.path contains:treePath] ||
-                       [treePath contains:record.path]));
+                  return (BOOL)((record.visible || includeHiddenWrites) &&
```

Comment:
😳  this condition is.... something. Probably fine to not touch it for safety reasons but yowza.